shell/shell.js

```diff
@@ -16,22 +16,153 @@
 
 const { classes: Cc, interfaces: Ci, results: Cr, utils: Cu } = Components;
 const { Runtime } = Cu.import('resource://qbrt/modules/Runtime.jsm', {});
+const { Services } = Cu.import('resource://gre/modules/Services.jsm', {});
 
-window.addEventListener('load', event => {
-  const browser = document.getElementById('content');
-  const url = window.arguments[0];
+let Accelerators = {
+  isMac: Services.appinfo.OS === 'Darwin',
+  keys: {
+    i: 73,
+    r: 82,
+    f5: 116,
+  },
+  Reload: function(event) {
+    const f5 = event.keyCode === this.keys.f5;
+    if (f5) {
+      return true;
+    }
+    if (this.isMac) {
+      // `Cmd + R`.
+      const cmdR = event.metaKey && event.keyCode === this.keys.r;
+      return cmdR;
+    }
+    // `Ctrl + Shift + R`.
+    const ctrlShiftR = event.ctrlKey && event.keyCode === this.keys.r;
+    return ctrlShiftR;
+  },
+  HardReload: function(event) {
+    // `Shift + F5`.
+    const shiftF5 = event.shiftKey && event.keyCode === this.keys.f5;
+    if (shiftF5) {
+      return true;
+    }
+    if (this.isMac) {
+      // `Cmd + Shift + R`.
+      const cmdShiftR = event.metaKey && event.shiftKey &&
+        event.keyCode === this.keys.r;
+      return cmdShiftR;
+    }
+    // `Ctrl + Shift + R`.
+    const ctrlShiftR = event.ctrlKey && event.shiftKey &&
+      event.keyCode === this.keys.r;
+    return ctrlShiftR;
+  },
+  ToggleDevTools: function(event) {
+    if (this.isMac) {
+      // `Cmd + Alt + I`.
+      const cmdAltI = event.metaKey && event.altKey &&
+        event.keyCode === this.keys.i;
+      return cmdAltI;
+    }
+    // `Ctrl + Shift + I`.
+    const ctrlShiftI = event.ctrlKey && event.shiftKey &&
+      event.keyCode === this.keys.i;
+    return ctrlShiftI;
+  },
+};
+
+function Shortcuts() {
```

Comment:
Do you expect _Shortcuts_ to be instantiated multiple times? It only gets instantiated once at the moment, but I imagine that could change, especially if it moves into a module that gets imported by multiple windows. (I'm just wondering if it makes more sense for it to be a singleton, like _Accelerators_ and _UI_, rather than a constructor.)